druntime/src/core/demangle.d

```diff
@@ -3273,3 +3273,68 @@ private struct BufSlice
     auto getSlice() inout nothrow scope { return buf[from .. to]; }
     size_t length() const scope { return to - from; }
 }
+
+
+
+/**
+ * C API to demangle D mangled names.
+ *
+ * Params:
+ *  mangled = The string to demangle.
+ *  buffer = A destination buffer.
+ *  bufferLength = The length of the destination buffer.
+ *
+ * Returns:
+ *  1 for success 0 for failure
+ */
+extern(C) int d_demangle(const(char*) mangled, char* buffer, size_t bufferLength)
+{
+    import core.stdc.string: strlen, memcpy;
+    if (mangled == null)
+        return 0;
+
+    auto mangledSlice = mangled[0 .. strlen(mangled)];
+    auto demangled = demangle(mangledSlice);
+
+    if (demangled.ptr == null || demangled.ptr == mangled)
+        return 0;
+
+    if (demangled.length > bufferLength)
+        return 0;
```

Comment:
you should return the length of the demangled name.
so the user can decide to allocate a bigger buffer if it does not fit.